File: ExpressionTreeTestObjects/CSCompiler/Literals.cs
using System;
using System.Collections.Generic;
using System.Linq.Expressions;
using static ExpressionTreeTestObjects.Functions;
using static ExpressionTreeTestObjects.Categories;

namespace ExpressionTreeTestObjects {
    partial class CSCompiler {
        [Category(Literal)]
        internal static readonly Expression True = Expr(() => true);

        [Category(Literal)]
        internal static readonly Expression False = Expr(() => false);

        [Category(Literal)]
        internal static readonly Expression Nothing = Expr(() => (string)null);

        [Category(Literal)]
        internal static readonly Expression Integer = Expr(() => 5);

        [Category(Literal)]
        internal static readonly Expression NonInteger = Expr(() => 7.32);

        [Category(Literal)]
        internal static readonly Expression String = Expr(() => "abcd");

        [Category(Literal)]
        internal static readonly Expression InterpolatedString = Expr(() => $"{new DateTime(2001, 1, 1)}");

        [Category(Literal)]
        internal static readonly Expression Type = Expr(() => typeof(string));
    }
}

File: ExpressionTreeTestObjects.VB/VBCompiler/Literals.vb
Partial Module VBCompiler
    <Category(Literal)>
    Friend [True] As Expression = Expr(Function() True)

    <Category(Literal)>
    Friend [False] As Expression = Expr(Function() False)

    <Category(Literal)>
    Friend NothingString As Expression = Expr(Function() CType(Nothing, String))

    <Category(Literal)>
    Friend [Nothing] As Expression = Expr(Function() Nothing)

    <Category(Literal)>
    Friend [Integer] As Expression = Expr(Function() 5)

    <Category(Literal)>
    Friend NonInteger As Expression = Expr(Function() 7.32)

    <Category(Literal)>
    Friend [String] As Expression = Expr(Function() "abcd")

    <Category(Literal)>
    Friend EscapedString As Expression = Expr(Function() """")

    <Category(Literal)>
    Friend ConstantNothingToObject As Expression = Expr(Function() Nothing)

    <Category(Literal)>
    Friend ConstantNothingToReferenceType As Expression = Expr(Of String)(Function() Nothing)

    <Category(Literal)>
    Friend ConstantNothingToValueType As Expression = Expr(Of Integer)(Function() Nothing)

    <Category(Literal)>
    Friend InterpolatedString As Expression = Expr(Function() $"{#1981-1-1#}")
End Module